datasets/pubmed/pubmed.py

```diff
@@ -358,28 +358,29 @@ def _generate_examples(self, filenames):
         """Yields examples."""
         id_ = 0
         for filename in filenames:
-            try:
-                tree = etree.parse(filename)
-                root = tree.getroot()
-                xmldict = self.xml_to_dictionnary(root)
-            except etree.ParseError:
-                logger.warning(f"Ignoring file {filename}, it is malformed")
-                continue
-
-            for article in xmldict["PubmedArticleSet"]["PubmedArticle"]:
-                self.update_citation(article)
-                new_article = default_article()
-
+            with open(filename, "rt", encoding="utf-8") as f:
+                xml_str = f.read()
                 try:
-                    deepupdate(new_article, article)
-                except Exception:
-                    logger.warning(f"Ignoring article {article}, it is malformed")
+                    root = etree.fromstring(xml_str)
+                    xmldict = self.xml_to_dictionnary(root)
+                except etree.ParseError:
+                    logger.warning(f"Ignoring file {filename}, it is malformed")
                     continue
 
-                try:
-                    _ = self.info.features.encode_example(new_article)
-                except Exception as e:
-                    logger.warning(f"Ignore example because {e}")
-                    continue
-                yield id_, new_article
-                id_ += 1
+                for article in xmldict["PubmedArticleSet"]["PubmedArticle"]:
+                    self.update_citation(article)
+                    new_article = default_article()
+
+                    try:
+                        deepupdate(new_article, article)
+                    except Exception:
+                        logger.warning(f"Ignoring article {article}, it is malformed")
+                        continue
+
+                    try:
+                        _ = self.info.features.encode_example(new_article)
+                    except Exception as e:
+                        logger.warning(f"Ignore example because {e}")
+                        continue
+                    yield id_, new_article
+                    id_ += 1
```

Comment:
I don't think this is necessary, once that `xml.etree.ElementTree.parse` already supports streaming. See:
- #3476

The only thing needs being changed is the import: `as ET` instead of `as etree`: https://github.com/huggingface/datasets/blob/71aa41caee3d2e9fde981cacc4a5fbddec58250d/datasets/pubmed/pubmed.py#L19